### ISSUE TITLE
🚀 Feature: incorporate publint rules

### ISSUE THREAD

**Issue by Zamiell** [JoshuaKGoldberg/eslint-plugin-package-json#85]
It would be nice if this plugin incorporated the rules from publint: https://publint.dev/rules

**Comment by JoshuaKGoldberg** [comment]
Agreed! I think a good next step would be to add publint to the table in #42 to see what isn't partially or fully duplicated in that table. Once that's done we can file issues for granular rules.

**Comment by JoshuaKGoldberg** [comment]
Agreed! I think a good next step would be to add publint to the table in #42 to see what isn't partially or fully duplicated in that table. Once that's done we can file issues for granular rules.

If you have time, that'd be fantastic. Otherwise I'll get to it... eventually 😅.

**Comment by michaelfaith** [comment]
@Zamiell are you still interested in some subset of those rules being considered for this plugin?

**Comment by Zamiell** [comment]
i think it would be nice to have these rules in eslint land to prevent people from having to use more than one tool to lint.
with that said, i personally don't currently use `eslint-plugin-package-json` in favor of `prettier-plugin-packagejson`.

**Comment by michaelfaith** [comment]
Sounds good.  If you have the time, as Josh mentioned here https://github.com/JoshuaKGoldberg/eslint-plugin-package-json/issues/85#issuecomment-1857149962, if you have any recommendations for specific rules that you think would be a good fit, we could add it to the tracking table.

**Comment by michaelfaith** [comment]
@Zamiell Closing this out, since there's not been any activity.  If you you're still interested in this, please follow-up with issues for the specific rules that you're interested in.

**Comment by michaelfaith** [comment]
> * 👍 [`FILE_INVALID_FORMAT`]

Pretty strongly disagree with adding this. Validating the nature of files referenced in a package.json is not something i think a lint rule should be responsible for.

> * 👍 [`FILE_DOES_NOT_EXIST`](https://publint.dev/rules#file_does_not_exist): 

Slightly against, but less strongly than above.

> * 👍 [`HAS_ESM_MAIN_BUT_NO_EXPORTS`](https://publint.dev/rules#has_esm_main_but_no_exports): 👍 a nice helper to migrate packages to an ESM-first -or at least ESM-dual- world

We already added `require-exports` in the publishable config, which is stricter than this and more black and white. I don't see the need for this.

> * 👍 [`EXPORTS_TYPES_SHOULD_BE_FIRST`](https://publint.dev/rules#exports_types_should_be_first): was tangentially discussed in [🚀 Feature: Enforce order of `imports` and `exports` #928](https://github.com/JoshuaKGoldberg/eslint-plugin-package-json/issues/928), but closed as no activity. I'd be in favor of re-thinking as this problem tripped me up a few times when I started on ESM overhauls.

In favor of some version of this. 👍

> * 👍 [`LOCAL_DEPENDENCY`](https://publint.dev/rules#local_dependency): not part of [`package-json/valid-dependencies`](https://github.com/JoshuaKGoldberg/eslint-plugin-package-json/blob/a766dbad7fe43fc9774e7b11d8265978d4b0ef1f/docs/rules/valid-dependencies.md), etc. - I'm thinking a good new rule

We already have an issue for this https://github.com/JoshuaKGoldberg/eslint-plugin-package-json/issues/1420

> * 👍 [`BIN_FILE_NOT_EXECUTABLE`](https://publint.dev/rules#bin_file_not_executable): not covered by [`package-json/valid-bin`](https://github.com/JoshuaKGoldberg/eslint-plugin-package-json/blob/a766dbad7fe43fc9774e7b11d8265978d4b0ef1f/docs/rules/valid-bin.md)

Not in favor of this. Projects like ours that are typescript, don't always have the bin file set up for execution as a source file. The package.json is going to be pointing at the built js file. And furthermore, in several cases, I've had the hash bang line added as a "banner" by rollup during the build, and a postbuild step flip the x perm to make it executable (i think pnpm even does this for you now). So I'm not sure what purpose a rule would have for us here. With a lot of these it's trying to use lint to solve problems that i don't think are lint's problem to solve.
